pkg/apis/eksctl.io/v1alpha5/validation_test.go

```diff
@@ -740,6 +740,17 @@ var _ = Describe("ClusterConfig validation", func() {
 					Expect(err).To(MatchError("Ipv6Cidr and Ipv6Pool must both be configured to use a custom IPv6 CIDR and address pool"))
 				})
 			})
+
+			When("its set alongside VPC.ID", func() {
+				It("returns an error", func() {
+					cfg.VPC.IPFamily = api.IPV6Family
+					cfg.VPC.IPv6Cidr = "foo"
+					cfg.VPC.IPv6Pool = "bar"
+					cfg.VPC.ID = "123"
+					err = cfg.ValidateVPCConfig()
+					Expect(err).To(MatchError("cannot provide VPC.IPv6Cidr when using a pre-existing VPC.ID"))
```

Comment:
Before matching the error, a good practice is to precede it with `Expect(err).To(HaveOccurred())`:
```suggestion
					Expect(err).To(HaveOccurred())
					Expect(err).To(MatchError("cannot provide VPC.IPv6Cidr when using a pre-existing VPC.ID"))
```

Reply:
I think thats no longer the case? I believe `MatchError` gracefully handles a nill error response

Reply:
yeah it handles it gracefully now:
```
    It("uses the existing VPC", func() {
      Expect(addErr).NotTo(HaveOccurred())
      Expect(addErr).To(MatchError("foo"))


• Failure [0.000 seconds]
Existing VPC
/home/jake/weave/eksctl/pkg/cfn/builder/vpc_existing_test.go:20
  CreateTemplate
  /home/jake/weave/eksctl/pkg/cfn/builder/vpc_existing_test.go:53
    uses the existing VPC [It]
    /home/jake/weave/eksctl/pkg/cfn/builder/vpc_existing_test.go:69

    Expected an error, got nil

    /home/jake/weave/eksctl/pkg/cfn/builder/vpc_existing_test.go:71

Reply:
>I believe MatchError gracefully handles a nill error response

I didn't mean to suggest that `MatchError` doesn't handle nil errors. I more meant that it's good for readability to ensure that an error has occurred before making any assertions on it.

Reply:
I don't really agree it improves readability tbh, in the past I used to do this pattern because `MatchError` didn't handle nil errors' gracefully, but since thats changed I don't think its worth it anymore